dss-apps/dss-datamodel-center/dss-datamodel-indicator-management/src/main/java/com/webank/wedatasphere/dss/datamodel/indicator/service/impl/IndicatorServiceImpl.java

```diff
@@ -0,0 +1,539 @@
+            }
+        }
+
+
+        Long orgId = org.getId();
+        String version = org.getVersion();
+
+        //更新原有指标
+        DssDatamodelIndicator updateOne = modelMapper.map(vo, DssDatamodelIndicator.class);
+        updateOne.setUpdateTime(new Date());
+        int result = getBaseMapper().update(updateOne, Wrappers.<DssDatamodelIndicator>lambdaUpdate().eq(DssDatamodelIndicator::getId, id));
+        if (result != 1) {
+            LOGGER.error("errorCode : {}, update indicator error not exists", ErrorCode.INDICATOR_UPDATE_ERROR.getCode());
+            throw new DSSDatamodelCenterException(ErrorCode.INDICATOR_UPDATE_ERROR.getCode(), "update indicator error not exists");
+        }
+
+        indicatorContentService.updateIndicatorContent(orgId, version, vo.getContent());
+
+
+
+        //同步atlas
+        UpdateModelTypeResult updateModelTypeResult = assertsSyncService.syncUpdateModel(
+                new UpdateModelEvent(this
+                        ,DataModelSecurityContextHolder.getContext().getDataModelAuthentication().getUser()
+                        ,vo.getFieldIdentifier()
+                        ,orgFieldIdentifier
+                        ,ClassificationConstant.INDICATOR)
+        );
+
+
+        return 1;
+    }
+
+
+    @Override
+    public int deleteIndicator(Long id) throws ErrorException {
+        DssDatamodelIndicator dssDatamodelIndicator = getBaseMapper().selectById(id);
+        if (dssDatamodelIndicator == null) {
+            throw new DSSDatamodelCenterException(ErrorCode.INDICATOR_DELETE_ERROR.getCode(), "indicator id " + id + " not exists");
+        }
+
+        //校验引用情况
+        if (datamodelReferencService.indicatorReferenceCount(dssDatamodelIndicator.getName())>0
+                ||datamodelReferencService.indicatorReferenceCount(dssDatamodelIndicator.getFieldIdentifier())>0){
+            throw new DSSDatamodelCenterException(ErrorCode.INDICATOR_DELETE_ERROR.getCode(), "indicator id " + id + " has referenced");
+        }
+
+
+//        //有版本不能删除
+//        if (indicatorVersionService.getBaseMapper().selectCount(Wrappers.<DssDatamodelIndicatorVersion>lambdaQuery().eq(DssDatamodelIndicatorVersion::getName, dssDatamodelIndicator.getName())) > 0) {
+//            throw new DSSDatamodelCenterException(ErrorCode.INDICATOR_DELETE_ERROR.getCode(), "indicator id " + id + " has version");
+//        }
+        //同时删除版本
+        indicatorVersionService.getBaseMapper()
+                .delete(Wrappers.<DssDatamodelIndicatorVersion>lambdaQuery().eq(DssDatamodelIndicatorVersion::getName, dssDatamodelIndicator.getName()));
+        //删除指标
+        getBaseMapper().deleteById(id);
+        indicatorContentService.getBaseMapper().delete(Wrappers.<DssDatamodelIndicatorContent>lambdaQuery().eq(DssDatamodelIndicatorContent::getIndicatorId, id));
+
+
+        //同步资产
+        assertsSyncService.syncDeleteModel(new DeleteModelEvent(this
```

Comment:
publisher.publishEvent(new DeleteModelEvent(this
                ,DataModelSecurityContextHolder.getContext().getDataModelAuthentication().getUser()
                ,dssDatamodelIndicator.getFieldIdentifier()
                ,ClassificationConstant.INDICATOR));